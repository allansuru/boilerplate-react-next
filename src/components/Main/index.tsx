import * as S from './styles'

const Main = ({
  title = 'React Avançado',
  description = 'TypeScript, ReactJS, NextJS e Styled Components',
  author = 'Allan Passos'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um átomo e React Avançado escrito ao lado."
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Author>by: {author}</S.Author>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para uma tela com código."
    />
  </S.Wrapper>
)
export default Main
