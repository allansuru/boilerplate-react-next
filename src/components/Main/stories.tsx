import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'title default',
    description: 'description default',
    author: 'author default'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: 'title basic',
  description: 'description basic',
  author: 'author default'
}

export const Default: Story = (args) => <Main {...args} />

export const RealLife: Story = (args) => <Main {...args} />
RealLife.args = {
  title: 'React Avançado',
  description: 'Estudos de React Avançado',
  author: 'Allan Passos'
}
