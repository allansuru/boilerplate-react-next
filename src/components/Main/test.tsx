import { render, screen } from '@testing-library/react'

import Main from './index'

describe('<Main />', () => {
  it('shoud render the heading', () => {
    expect(screen.getByRole('heading', { name: /react avan√ßado/i }))
      .toBeInTheDocument
  })
})
