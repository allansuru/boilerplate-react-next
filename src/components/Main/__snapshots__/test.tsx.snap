// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> shoud render colors correct 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 dMywMa"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="styles__Logo-dzx1ao-1 dcISKB"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2 kSTVFj"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 edoRUQ"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <h3
    class="styles__Author-dzx1ao-4 iZwehP"
  >
    by: 
    Allan Passos
  </h3>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="styles__Illustration-dzx1ao-5 eOjuXo"
    src="/img/hero-illustration.svg"
  />
</main>
`;

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 dMywMa"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="styles__Logo-dzx1ao-1 dcISKB"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2 kSTVFj"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 edoRUQ"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <h3
    class="styles__Author-dzx1ao-4 iZwehP"
  >
    by: 
    Allan Passos
  </h3>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="styles__Illustration-dzx1ao-5 eOjuXo"
    src="/img/hero-illustration.svg"
  />
</main>
`;
